### PR TITLE
Support selection of histogram type facets

### PR DESCRIPTION
**Pull request by bfcrampton**

### CODE REVIEW

File: ui/src/components/facets/HistogramFacet.js [hunk 1]
```diff
@@ -2,8 +2,6 @@ import React, { Component } from "react";
 import VegaLite from "react-vega-lite";
```
Comment (melissachang) [inline ui/src/components/facets/HistogramFacet.js]: Looks like we're only using react-vega-lite and vega-tooltip. Can you delete everything from package.json (react-vega, vega-embed, vega-lib, vega-lite)?

File: ui/src/components/facets/HistogramFacet.js [hunk 2]
```diff
@@ -88,6 +95,35 @@ class FacetHistogram extends Component {
       !this.props.selectedValues.includes(facetValue.name)
     );
   }
+
+  onClick(event, item) {
+    // Ignore clicks which are not located on histogram
+    // bars.
+    if (item != null) {
```
Comment (melissachang) [inline ui/src/components/facets/HistogramFacet.js]:
```
if (item && item.datum) {
```
If I click inside the graph area but not on a bar, item is set and item.datum is not set.

File: ui/src/components/facets/HistogramFacet.js [hunk 3]
```diff
@@ -88,6 +95,35 @@ class FacetHistogram extends Component {
...
+      // facetValue is a string, eg "female"
+      const facetValue = item.datum.facet_value;
+      let isSelected;
+      if (
+        this.props.selectedValues != null &&
```
Comment (melissachang) [inline ui/src/components/facets/HistogramFacet.js]: // this.props.selectedValues contains what was selected before the click.
// isSelected contains if facet value was selected after the click.

File: ui/src/components/facets/FacetsGrid.js [hunk 4]
```diff
@@ -17,6 +17,7 @@ function facetCard(props, facet) {
     return (
       <HistogramFacet
         facet={facet}
+        updateFacets={props.updateFacets}
```
Comment (melissachang) [inline ui/src/components/facets/FacetsGrid.js]: This has the same problem as the tooltip -- it's not possible to select values with low numbers. Maybe if we do the transparent background bar, that could help with this.

File: ui/src/components/facets/HistogramFacet.js (hunk 1 above)
Comment (bfcrampton) [reply ui/src/components/facets/HistogramFacet.js]: Done.

File: ui/src/components/facets/HistogramFacet.js (hunk 2 above)
Comment (bfcrampton) [reply ui/src/components/facets/HistogramFacet.js]: Done.

File: ui/src/components/facets/HistogramFacet.js (hunk 3 above)
Comment (bfcrampton) [reply ui/src/components/facets/HistogramFacet.js]: Done.

File: ui/src/components/facets/FacetsGrid.js (hunk 4 above)
Comment (bfcrampton) [reply ui/src/components/facets/FacetsGrid.js]: Noted, I'll fix both in my next PR.

**Comment by bfcrampton** [comment]
It turns out that since react-vega-lite has a [peerDependency](https://nodejs.org/en/blog/npm/peer-dependencies/#using-peer-dependencies) [on vega-lite](https://github.com/vega/react-vega-lite/blob/master/package.json#L16), and react-vega [on vega-lib](https://github.com/vega/react-vega/blob/master/package.json#L16), they also have to be present on our `package.json`. react-vega-lite has a direct dependency on react-vega though so kept that out.

File: ui/package.json [hunk 5]
```diff
@@ -13,11 +13,10 @@
     "react-dom": "^16.3.0",
     "react-scripts": "^2.0.0",
     "react-select": "^2.0.0",
-    "react-vega": "^4.0.2",
     "react-vega-lite": "^2.0.2",
-    "vega-embed": "^3.29.0",
     "vega-lib": "^4.4.0",
-    "vega-lite": "^3.0.0-rc12"
+    "vega-lite": "^3.0.0-rc12",
```
Comment (melissachang) [inline ui/package.json]: Add a comment for why these two are here, so someone else in the future doesn't make the same mistake. See https://stackoverflow.com/a/14221781/6447189

File: ui/package.json [hunk 6]
```diff
@@ -46,5 +45,11 @@
   },
   "jest": {
     "preset": "jest-puppeteer"
-  }
+  },
+  "browserslist": [
```
Comment (melissachang) [inline ui/package.json]: Why did this change?

File: ui/package.json (hunk 5 above)
Comment (bfcrampton) [reply ui/package.json]: Done.

File: ui/package.json (hunk 6 above)
Comment (bfcrampton) [reply ui/package.json]: Huh, not sure. Reverted.

File: ui/package.json [hunk 7]
```diff
@@ -29,6 +28,7 @@
     "eject": "react-scripts eject"
   },
   "//": [
+    "vega-lib and vega-lite are peerDependencies of react-vega-lite.",
```
Comment (melissachang) [inline ui/package.json]: Is it possible to split this into two comments? Leave 32 here, move 31 to 17.

Comment (bfcrampton) [reply ui/package.json]: Yup it is, done.

**Pull request merged**